bls/src/main/java/org/dashj/bls/CoreMPL.java

```diff
@@ -0,0 +1,103 @@
+/* ----------------------------------------------------------------------------
+ * This file was automatically generated by SWIG (http://www.swig.org).
+ * Version 4.0.2
+ *
+ * Do not make changes to this file unless you know what you are doing--modify
+ * the SWIG interface file instead.
+ * ----------------------------------------------------------------------------- */
+
+package org.dashj.bls;
+
+public class CoreMPL {
+  private transient long swigCPtr;
+  protected transient boolean swigCMemOwn;
+
+  protected CoreMPL(long cPtr, boolean cMemoryOwn) {
+    swigCMemOwn = cMemoryOwn;
+    swigCPtr = cPtr;
+  }
+
+  protected static long getCPtr(CoreMPL obj) {
+    return (obj == null) ? 0 : obj.swigCPtr;
+  }
+
+  @SuppressWarnings("deprecation")
+  protected void finalize() {
+    delete();
+  }
+
+  public synchronized void delete() {
+    if (swigCPtr != 0) {
+      if (swigCMemOwn) {
+        swigCMemOwn = false;
+        DASHJBLSJNI.delete_CoreMPL(swigCPtr);
+      }
+      swigCPtr = 0;
+    }
+  }
+
+  public CoreMPL(String strId) {
+    this(DASHJBLSJNI.new_CoreMPL(strId), true);
+  }
+
+  public PrivateKey keyGen(byte[] seed) {
+    return new PrivateKey(DASHJBLSJNI.CoreMPL_keyGen(swigCPtr, this, seed), true);
+  }
+
+  public byte[] skToPk(PrivateKey seckey) { return DASHJBLSJNI.CoreMPL_skToPk(swigCPtr, this, PrivateKey.getCPtr(seckey), seckey); }
+
+  public G1Element skToG1(PrivateKey seckey) {
+    return new G1Element(DASHJBLSJNI.CoreMPL_skToG1(swigCPtr, this, PrivateKey.getCPtr(seckey), seckey), true);
+  }
```

Comment:
each class method is wrapping a native method in DASHJBLSJNI.  `swigCPtr` is `this` in C++.